resources/git-rebase.md

```diff
@@ -0,0 +1,66 @@
+# Git rebase
+
+## What’s rebasing?
+
+Git rebase takes a branch you select and merges it on to the branch where you’re currently working. Then it takes all of your local commits and applies them on top.
+
+## You said it merges the branches. Why not use `git merge`?
+
+Git rebase solves the same issues as git merge, but git merge will make a merge commit every time you pull upstream changes. This can make your git history really noisy, really fast, making it difficult for you and your team to understand the history of your project.
+
+## When should I rebase?
+
+One common reason for rebasing is to get changes from another branch and apply them to your own. Let’s say you and your team are busy working on your project. You’re coding on your local branch when someone else on your team finishes a feature and merges it to master. The new code your friend just pushed is a change you want on your local branch. You can use `git rebase <nameOfYourFriendsBranch>` to add these changes to your local branch and keep working.
+
+Another reason to rebase is to head off any potential merge conflicts. A merge conflict occurs when your branch and another branch have code edits on the same lines. Git doesn’t know which of the edits should be kept, so it throws a conflict that a human (you!) has to resolve.
+
+## Merge conflicts sound terrible. How can I make sure I never have to deal with them?
+
+Alas, dealing with merge conflicts is a part of developer life. But they don’t have to be awful! To mitigate conflicts, commit your work often and rebase master when new code is added. If everyone on your team does this, you’ll keep your local branches up-to-date and lower the entire team's odds of having to fix a large merge conflict.
+
+Using the interactive mode when performing a rebase helps as well. Interactive mode can be used by adding the -i flag to your rebase command like so:
+
+ `git rebase -i <nameOfBranchToRebaseIn>`
+
+Using interactive mode when rebasing lets you see the commits that you're rebasing, and allows you to edit and squash commits.
+
+## What is git squash?
+Git squash allows you to take several commits and merge them into a single commit. Say you've finished a feature - styling a nav bar - and your git history looks like this:
```

Comment:
To avoid saying "merge" often, what about using the word "combine"?

Reply:
good call. updated.